src/pages/index.js

```diff
@@ -18,42 +19,201 @@ import {
   profileEditForm,
   addCardForm,
   previewCardModal,
+  profileAvatarButton,
+  profileAvatarModal,
+  profileAvatarForm,
+  cardDeleteButton,
+  cardDeleteModal,
+  cardDeleteForm,
 } from "../utils/constants.js";
+import PopupWithConfirmation from "../components/PopupWithConfirmation.js";
 
-const section = new Section(
-  {
-    items: initialCards,
-    renderer: (cardData) => {
-      const sectionCard = createCard(cardData);
-      section.addItem(sectionCard);
-    },
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "7de0cfd0-91fd-49fd-9d21-140e9d9b5161",
+    "Content-Type": "application/json",
   },
-  ".cards__list "
-);
-section.renderItems();
+});
+
+let section;
+
+api
+  .getInitialCards()
+  .then((cards) => {
+    section = new Section(
+      {
+        items: cards,
+        renderer: (cardData) => {
+          const sectionCard = createCard(cardData);
+          section.addItem(sectionCard);
+        },
+      },
+      ".cards__list "
+    );
+
+    section.renderItems();
+  })
+  .catch((error) => {
+    console.error("Error fetching initial cards", error);
+  });
+
+api
+  .getUserInfo()
+  .then((userData) => {
+    userInfo.setUserInfo({
+      name: userData.name,
+      description: userData.about,
+      avatar: userData.avatar,
+    });
+  })
+  .catch((err) => {
+    console.log(err);
+  });
+
 // User Info
-const userInfo = new UserInfo(".profile__title", ".profile__description");
+const userInfo = new UserInfo({
+  nameSelector: ".profile__title",
+  descriptionSelector: ".profile__description",
+  avatarSelector: ".profile__image",
+});
+
+// Function to handle click event
+function handleImageClick({ name, link }) {
+  popupWithImage.open(name, link);
+}
+
+function createCard(cardData) {
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleDeleteCard,
+    handleLike
+  );
+  return card.getView();
+}
+
+function handleProfileEditSubmit({ name, description }) {
+  profileEditPopup.setLoading(true);
+  api
+    .updateUserInfo(name, description)
+    .then(() => {
+      userInfo.setUserInfo({ name: name, description: description });
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      profileEditPopup.setLoading(false);
+      profileEditPopup.close();
```

Comment:
* You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well


**Please, correct it everywhere**